exercises/hangman/HangmanTest.cs

```diff
@@ -1,135 +1,298 @@
-﻿using Xunit;
-
-public class HangmanTest
+﻿using System;
+using Xunit;
+using Microsoft.Reactive.Testing;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reactive.Concurrency;
+
+public class HangmanTests: ReactiveTest
 {
     [Fact]
-    public void Initially_9_failures_are_allowed()
+    public void Initial_state_masks_the_word()
     {
-        var game = new HangmanGame("foo");
+        var hangman = new Hangman("foo");
+        var actual = "";
+
+        // +a->
+        hangman.StateObservable.Subscribe(
+            x => actual = x.MaskedWord,
+            ex => throw new Exception("Should not finish with too many tries"),
+            () => throw new Exception("Should not win yet"));
+        Assert.Equal("___", actual);
+    }
 
-        HangmanState lastState = null;
-        game.StateChanged += (sender, state) => lastState = state;
+    [Fact(Skip = "Remove to run test")]
+    public void Initial_state_have_9_failures_allowed()
```

Comment:
```suggestion
    public void Initial_state_has_9_remaining_guesses()
```

I feel that this better corresponds to the code (it checks the `RemainingGuesses` property).